Pidaq/Lib/can_manager/can_manager.py

```diff
@@ -0,0 +1,150 @@
+"""Contains classes and methods to implement and manage the can (socketcan) bus
+
+Classes:
+    CanManager
+    SensorReading
+
+Functions:
+    parse_conversion_factor
+"""
+import json
+import os
+
+import can
+
+
+class CanManager:
+    """Manages the can bus, and sets the SensorReading objects for the test being run
+
+    This class contains the socketcan bus instance that will be used to communicate with
+    the sensor board, as well as the configured sensor readings for the particular project
+    that is being tested
+
+    Attributes:
+        bus (can.interfaces.socketcan.SocketcanBus)
+        messages (dict): dict of all the messages to be expected from the
+                         sensor board. Keys are message ids, and values are
+                         SensorReading objects
+        message_ids (list(int)): list of all the message ids, hexidecimal integers
+
+    Methods:
+        read_message_config(project: str)
+            Reads the configuration of sensor readings from messageconfig.json
+        send_message(id: int, data: list)
+        read_bus
+            Reads the bus for a message
+        assign_message_data(bus_message: can.Message)
+            assigns data to the correct SensorReading object
+    """
+
+    def __init__(self, bus_name: str) -> None:
+        self.bus = can.interfaces.socketcan.SocketcanBus(channel=bus_name)
+        self.messages = {}
+        self.message_ids = []
+
+    def read_message_config(self, project: str, config_file: str, path=None) -> None:
+        """Reads sensor readings configuration from messageconfig.json
+
+        Constructs SensorReading objects for all expected sensor readings,
+        and appends them to a list
+
+        Parameters:
+            project (str): current project, must be one of dts, suspension, or windtunnel
+            config_file (str): message configuration file name
+            path (str): path to folder containing configuration file, if no path specified, current working
+                        directory is assumed
+        """
+        if project.lower() not in ["dts", "suspension", "windtunnel"]:
+            raise ValueError(f'Error: {project} is not a valid project')
+        file_path = os.getcwd() if path is None else path
+        with open(os.path.join(file_path, config_file), 'r+') as config:
+            config_dict = json.load(config)
+            for reading in config_dict[project.lower()]['readings']:
+                message_id = reading['message_id']
+                self.messages[message_id] = SensorReading(
+                    project,
+                    reading['message_id'],
+                    reading['reading'],
+                    reading['conversion_factor'],
+                )
+                self.message_ids.append(int(reading['message_id'], 16))
```

Comment:
No longer using hex strings, don't need this conversion anymore.

Also discussed removing the message_ids and using the dict instead